src/main/java/org/betonquest/betonquest/database/Connector.java

```diff
@@ -49,14 +49,19 @@ public Connector() {
      * This method should be used before any other database operations.
      */
     @SuppressFBWarnings({"ODR_OPEN_DATABASE_RESOURCE", "OBL_UNSATISFIED_OBLIGATION"})
-    public final void refresh() {
-        try {
-            connection.prepareStatement("SELECT 1").executeQuery().close();
-        } catch (final SQLException e) {
-            log.error("Reconnecting to the database", e);
-            database.closeConnection();
+    public final boolean refresh() {
```

Comment:
In the constructor of this class only `refresh()` should be called.

Reply:
The connector is created in several other classes. I could pass the parameter to the class, but I don't see the advantage of it at the moment. Would you like to explain this in more detail?

Reply:
Oh, I see this was written in a bad way. It is about the `database.getConnection()` call, there's no reason to call that first and then refresh immediately.